### PR TITLE
docs: simplify Docker toolchain docs (#9966)

### LINKED ISSUE #9966: TOB-ARGO-024: Toolchain docs should link to Docker docs about not the `docker` group
# Summary

The [toolchain docs](https://argo-cd.readthedocs.io/en/stable/developer-guide/toolchain-guide/#before-you-start) say "don't use root w/ Docker," but it doesn't explain _why_. 

# Motivation

People sometimes ignore security recommendations unless they understand _why_ the recommendations were made.

This was pointed out in item 24 of the Trail of Bits security audit. https://github.com/argoproj/argoproj/blob/master/docs/argo_security_final_report.pdf

# Proposal

Link to the Docker docs explaining why running Docker as root is a bad idea.

### PR DESCRIPTION
**Pull request by crenshaw-dev**
The information about `root` is unnecessary and potentially confusing.

Fixes #9966

### CODE REVIEW

File: docs/developer-guide/toolchain-guide.md [hunk 1]
```diff
@@ -24,8 +24,7 @@ You will need at least the following things in your toolchain in order to develo
 
 * A Kubernetes cluster. You won't need a fully blown multi-master, multi-node cluster, but you will need something like K3S, Minikube or microk8s. You will also need a working Kubernetes client (`kubectl`) configuration in your development environment. The configuration must reside in `~/.kube/config` and the API server URL must point to the IP address of your local machine (or VM), and **not** to `localhost` or `127.0.0.1` if you are using the virtualized development toolchain (see below)
 
-* You will also need a working Docker runtime environment, to be able to build and run images.
-The Docker version must be fairly recent, and support multi-stage builds. You should not work as root. Make your local user a member of the `docker` group to be able to control the Docker service on your machine.
+* You will also need a working Docker runtime environment, to be able to build and run images. The Docker version must 17.05.0 or higher, to support multi-stage builds.
```
Comment (Kerwood) [inline docs/developer-guide/toolchain-guide.md:27]: Missing a "be".
> The Docker version must **be** 17.05.0 or higher

Comment (crenshaw-dev) [reply docs/developer-guide/toolchain-guide.md]: Thanks!

**Comment by crenshaw-dev** [comment]
Markdown 3.4 broke the ReadTheDocs build. https://pypi.org/project/Markdown/#history

**Review by simster7 — APPROVED** [review]

**Review by terrytangyuan — APPROVED** [review]
LGTM

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for 2.4.7.

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.2 for 2.2.12 and release-2.3 for 2.3.7.

**Pull request merged**